internal/querynode/task.go

```diff
@@ -114,6 +114,23 @@ type releasePartitionsTask struct {
 	node *QueryNode
 }
 
+// getSegmentInfosFromWatchDmChannelsRequest returns the segment infos exist in the watchDmChannelsRequest
+// this method is used to retrieve the segment infos needed by a vChannel.
+// it is expected all segmentId should exist in WatchDmChannelsRequest.SegmentInfos,
+// including flushed, unflushed and dropped.
+func getSegmentInfosFromWatchDmChannelsRequest(segmentIds []int64, watchDmChannelsRequest *queryPb.WatchDmChannelsRequest) ([]*datapb.SegmentInfo, error) {
```

Comment:
do we need this call, might be just use. watchDmChannelsRequest.SegmentInfos[id] because I don't see anything else on it rather than error handling

Reply:
Yes, it is not quite necessary, removed it.